libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/RestClient.swift

```diff
@@ -0,0 +1,432 @@
+    func fileShares(_ sfdcId: String, page: Int, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+       let request = self.request(forFileShares: sfdcId, page: UInt(page), apiVersion: self.apiVersion)
+       self.send(request: request, completionBlock)
+       return request
+    }
+    
+    /// Add a File Share
+    /// - Parameter fileId: File identifier.
+    /// - Parameter entityId: Entity identifier
+    /// - Parameter shareType: The type of Share.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func addFileShare(_ fileId: String, entityId: String, shareType: String, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+       let request = self.request(forAddFileShare: fileId, entityId: entityId, shareType: shareType, apiVersion: self.apiVersion)
+       self.send(request: request, completionBlock)
+       return request
+    }
+    
+    /// Delete a file share.
+    /// - Parameter shareId: Share identtifier.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func deleteFileShare(_ shareId: String, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+       let request = self.request(forDeleteFileShare: shareId, apiVersion: self.apiVersion)
+       self.send(request: request, completionBlock)
+       return request
+    }
+    
+    /// Upload a File.
+    /// - Parameter data: Data contents.
+    /// - Parameter name: Name for the file
+    /// - Parameter description: Description for file.
+    /// - Parameter mimeType: The mime type.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func uploadFileShare(_ data: Data, name: String, description: String, mimeType: String, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.request(forUploadFile: data, name: name, description: description, mimeType: mimeType, apiVersion: self.apiVersion)
+       self.send(request: request, completionBlock)
+       return request
+    }
+    
+    /// Upload a photo.
+    /// - Parameter data: Data contents.
+    /// - Parameter name: Name for the file
+    /// - Parameter description: Description for file.
+    /// - Parameter mimeType: The mime type.
+    /// - Parameter userId: Identifier for user.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func uploadProfilePhoto(_ data: Data, name: String, description: String, mimeType: String, userId: String, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.request(forProfilePhotoUpload: data, fileName: name, mimeType: mimeType, userId: userId, apiVersion: self.apiVersion)
+       self.send(request: request, completionBlock)
+       
+       return request
+    }
+ 
+}
+
+@available(iOS 13.0, tvOS 13.0, watchOS 6.0, *)
```

Comment:
Do we need `tvOS` and `watchOS` declarations here? We don't support either of those.

Reply:
I think we should keep the restrictions in place for watchOS.